### PR TITLE
Add dependabot for Gradle dependencies

### PR DESCRIPTION
**Pull request by blasten**
Adds dependabot for Gradle dependencies.

Unfortunately, dependabot doesn't support globs patterns in directory.

cc @stuartmorgan

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
+version: 2
+updates:
+  - package-ecosystem: "gradle"
+    directory: "/packages/flutter_plugin_android_lifecycle/example/android/app"
+    schedule:
+      interval: "weekly"
```
Comment (domesticmouse) [inline .github/dependabot.yml:6]: FWIW, I'd suggest making the interval daily, just to spread out the dependabot PRs

Comment (blasten) [reply .github/dependabot.yml]: sgtm. PTAL

**Review by domesticmouse — APPROVED** [review]

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,139 @@
+version: 2
+updates:
+  - package-ecosystem: "gradle"
+    directory: "/packages/flutter_plugin_android_lifecycle/example/android/app"
```
Comment (stuartmorgan) [inline .github/dependabot.yml:4]: Will checking the examples transitively check and update the plugin dependencies themselves?

File: .github/dependabot.yml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    open-pull-requests-limit: 10
+
+  - package-ecosystem: "gradle"
+    directory: "/packages/path_provider/path_provider_android/example/android/app"
+    schedule:
+      interval: "daily"
+    open-pull-requests-limit: 10
+
+  - package-ecosystem: "gradle"
+    directory: "/packages/webview_flutter/webview_flutter/example/android/app"
```
Comment (stuartmorgan) [inline .github/dependabot.yml:22]: We should order this file alphabetically by directory so we can easily see what's there.

Comment (stuartmorgan) [inline .github/dependabot.yml:22]: Speaking of which, the `_android` package is missing here.

File: .github/dependabot.yml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    schedule:
+      interval: "daily"
+    open-pull-requests-limit: 10
+
+  - package-ecosystem: "gradle"
+    directory: "/packages/connectivity/connectivity/example/android/app"
```
Comment (stuartmorgan) [inline .github/dependabot.yml:28]: This plugin doesn't exist any more.

File: .github/dependabot.yml [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    open-pull-requests-limit: 10
+
+  - package-ecosystem: "gradle"
+    directory: "/packages/local_auth/local_auth/example/android/app"
+    schedule:
+      interval: "daily"
+    open-pull-requests-limit: 10
+
+  - package-ecosystem: "gradle"
+    directory: "/packages/video_player/video_player_android/example/android/app"
```
Comment (stuartmorgan) [inline .github/dependabot.yml:64]: The app-facing package is missing.

With only a couple of exceptions, there should be two examples for every plugin.

File: .github/dependabot.yml [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    open-pull-requests-limit: 10
+
+  - package-ecosystem: "gradle"
+    directory: "/packages/espresso/example/android/app"
+    schedule:
+      interval: "daily"
+    open-pull-requests-limit: 10
+
+  - package-ecosystem: "gradle"
+    directory: "/packages/espresso/example/android/app"
```
Comment (stuartmorgan) [inline .github/dependabot.yml:136]: This is a duplicate entry.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Nice!

Should we put a `commit-message` entry to do the `[plugin_name]` prefix?

File: .github/dependabot.yml (hunk 2 above)
Comment (blasten) [reply .github/dependabot.yml:4]: I don't think so. It's not very smart from what I can tell

File: .github/dependabot.yml (hunk 3 above)
Comment (blasten) [reply .github/dependabot.yml:22]: done

Comment (blasten) [reply .github/dependabot.yml:22]: done

File: .github/dependabot.yml (hunk 4 above)
Comment (blasten) [reply .github/dependabot.yml:28]: done

File: .github/dependabot.yml (hunk 5 above)
Comment (blasten) [reply .github/dependabot.yml:64]: done

File: .github/dependabot.yml (hunk 6 above)
Comment (blasten) [reply .github/dependabot.yml:136]: done

File: .github/dependabot.yml [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+    open-pull-requests-limit: 10
+
+  - package-ecosystem: "gradle"
+    directory: "/packages/url_launcher/url_launcher/example/android/app"
+    commit-message:
+      prefix: "[url_launcher]"
+    schedule:
+      interval: "daily"
+    open-pull-requests-limit: 10
+
```
Comment (stuartmorgan) [inline .github/dependabot.yml:218]: The app-facing package's example is missing.

File: .github/dependabot.yml [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+  - package-ecosystem: "gradle"
+    directory: "/packages/path_provider/path_provider/example/android/app"
+    commit-message:
+      prefix: "[path_provider]"
+    schedule:
+      interval: "daily"
+    open-pull-requests-limit: 10
+
+  - package-ecosystem: "gradle"
+    directory: "/packages/path_provider/path_provider_android/example/android/app"
```
Comment (stuartmorgan) [inline .github/dependabot.yml:148]: The plugin is missing.

File: .github/dependabot.yml [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+  - package-ecosystem: "gradle"
+    directory: "/packages/in_app_purchase/in_app_purchase/example/android/app"
+    commit-message:
+      prefix: "[in_app_purchase]"
+    schedule:
+      interval: "daily"
+    open-pull-requests-limit: 10
+
+  - package-ecosystem: "gradle"
+    directory: "/packages/image_picker/image_picker/example/android/app"
```
Comment (stuartmorgan) [inline .github/dependabot.yml:108]: This is missing both parts of the implementation package.

File: .github/dependabot.yml [hunk 10]
```diff
@@ -0,0 +1,257 @@
...
+  - package-ecosystem: "gradle"
+    directory: "/packages/camera/camera/example/android/app"
+    commit-message:
+      prefix: "[camera]"
+    schedule:
+      interval: "daily"
+    open-pull-requests-limit: 10
+
+  - package-ecosystem: "gradle"
+    directory: "/packages/espresso/example/android/app"
```
Comment (stuartmorgan) [inline .github/dependabot.yml:20]: The plugin is missing.

**Review by stuartmorgan — APPROVED** [review]
LGTM with more added. I'm not sure I caught everything though.

Given that this is fiddly to get right, and will silently get out of sync with changes (e.g., federating the last couple of plugins), I think I should make tooling that verifies dependabot coverage. It'll be useful when we enable Dart dependabot too.

I'll file an issue for that. (Doesn't need to block yours though).

File: .github/dependabot.yml (hunk 7 above)
Comment (blasten) [reply .github/dependabot.yml:218]: url_launcher doesn't have an app-facing package. https://github.com/flutter/plugins/tree/main/packages/url_launcher/url_launcher

File: .github/dependabot.yml (hunk 8 above)
Comment (blasten) [reply .github/dependabot.yml:148]: done

File: .github/dependabot.yml (hunk 9 above)
Comment (blasten) [reply .github/dependabot.yml:108]: done

File: .github/dependabot.yml (hunk 10 above)
Comment (blasten) [reply .github/dependabot.yml:20]: done

**Comment by ditman** [comment]
This has failed [in master](https://github.com/flutter/plugins/runs/6266715349) with the following:


## Your .github/dependabot.yml contained invalid details

Dependabot encountered the following error when parsing your .github/dependabot.yml:

```
The property '#/updates/4/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/5/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/6/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/7/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/8/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/9/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/10/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/11/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/12/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/13/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/15/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/16/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/17/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/18/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/21/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/22/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/23/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/24/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/26/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/27/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/28/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/29/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/31/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/32/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/33/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/34/commit-message/prefix' was not of a maximum string length of 15
The property '#/updates/35/commit-message/prefix' was not of a maximum string length of 15
```

Please update the config file to conform with Dependabot's specification.

For more info on the config file format, see [the config file documentation](https://docs.github.com/en/github/administering-a-repository/configuration-options-for-dependency-updates#configuration-options-for-dependabotyml)

**Comment by ditman** [comment]
Cool way of regexing the baddies:

```regex
prefix: "\[[^\]]{14,}\]"
```

**Pull request merged**